app/models/rapids/occupation_standard.rb

```diff
@@ -0,0 +1,53 @@
+module RAPIDS
+  class OccupationStandard
+    OCC_TYPE_MAPPING = {
+      "Time-Based" => :time,
+      "Competency-Based" => :competency,
+      "Hybrid" => :hybrid
+    }
+
+    class << self
+      def initialize_from_response(response)
+        ::OccupationStandard.new(
+          title: fix_encoding(response["occupationTitle"]),
+          onet_code: response["onetSocCode"],
+          rapids_code: response["rapidsCode"],
+          ojt_type: ojt_type(response["occType"]),
+          registration_agency: find_registration_agency_by_sponsor_number(
+            response["sponsorNumber"]
+          )
+        )
+      end
+
+      private
+
+      def ojt_type(occ_type)
+        OCC_TYPE_MAPPING[occ_type]
+      end
+
+      def find_registration_agency_by_sponsor_number(sponsor_number)
+        registration_agency = nil
+
+        if sponsor_number.present?
+          parsed_sponsor_name = sponsor_number.match(/-?(?<state>[A-Z]{2})/)
+
+          state_abbreviation = parsed_sponsor_name && parsed_sponsor_name[:state]
+
+          state = State.find_by(
+            abbreviation: state_abbreviation
+          )
+
+          if state
+            registration_agency = state.registration_agencies.last
```

Comment:
Some states have both OA and SAA -- do we know which one we would want to grab in those cases?

Reply:
@littleforest didn't know that case was possible 😮  I'll ask Patrick

Reply:
It must use OA. Code updated to reflect this